NuGet outputs a bunch of warnings about duplicate imports during restore

Build: 4.0.0.2215
VS: 26114.2 (D15Rel)

From a Visual Studio command-prompt:

1. `git clone http://github.com/donet/roslyn-project-system.git`
2. `git checkout 9f8cbdb0ca876ae037bb3e3fb0dd1bc1e70cf880`
3. Grab NuGet.exe 4.0.0.2215 and replace build\bin\nuget.exe
4. `build`

Expected: No warnings
Actual: About 50 warnings with:

```
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Microsoft.Common.targets\ImportAfter\Microsoft.NuGet.ImportAfter.targets(22,3): warning MSB4011: "C:\Users\davkean\AppData\Local\Temp\NuGet-Scratch\gpvksigy.qfp.targets" cannot be imported again. It was already imported at "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\bin\Microsoft.Common.CurrentVersion.targets (5679,3)". This is most likely a build authoring error. This subsequent import will be ignored. [E:\project-system\src\DeployTestDependencies\DeployTestDependencies.csproj] [E:\project-system\build\build.proj]
```

--- a/src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs
+++ b/src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs
@@ -191,7 +191,8 @@ namespace NuGet.CommandLine
                     UseShellExecute = false,
                     FileName = msbuildPath,
                     Arguments = argumentBuilder.ToString(),
-                    RedirectStandardError = true
+                    RedirectStandardError = true,
+                    RedirectStandardOutput = true
                 };
 
                 console.LogDebug($"{processStartInfo.FileName} {processStartInfo.Arguments}");
@@ -199,7 +200,10 @@ namespace NuGet.CommandLine
                 using (var process = Process.Start(processStartInfo))
                 {
                     var errors = new StringBuilder();
-                    var errorTask = ConsumeStreamReaderAsync(process.StandardError, errors);
+                    var output = new StringBuilder();
+                    var excluded = new string[] { "msb4011", entryPointTargetPath };
+                    var errorTask = ConsumeStreamReaderAsync(process.StandardError, errors, filter: null);
+                    var outputTask = ConsumeStreamReaderAsync(process.StandardOutput, output, filter: (line) => IsIgnoredOutput(line, excluded));
                     var finished = process.WaitForExit(timeOut);
                     if (!finished)
                     {
@@ -218,6 +222,8 @@ namespace NuGet.CommandLine
                         throw new CommandLineException(
                             LocalizedResourceManager.GetString(nameof(NuGetResources.Error_MsBuildTimedOut)));
                     }
+
+                    await outputTask;
 
                     if (process.ExitCode != 0)
                     {
@@ -242,14 +248,23 @@ namespace NuGet.CommandLine
             }
         }
 
-        private static async Task ConsumeStreamReaderAsync(StreamReader reader, StringBuilder lines)
+        private static bool IsIgnoredOutput(string line, string[] excluded)
+        {
+            return excluded.All(p => line.IndexOf(p, StringComparison.OrdinalIgnoreCase) >= 0);
+        }
+
+        private static async Task ConsumeStreamReaderAsync(StreamReader reader, StringBuilder lines, Func<string, bool> filter)
         {
             await Task.Yield();
 
             string line;
             while ((line = await reader.ReadLineAsync()) != null)
             {
-                lines.AppendLine(line);
+                if (filter == null || 
+                    !filter(line))
+                {
+                    lines.AppendLine(line);
+                }
             }
         }
 

